e2e/wdio/platform/tests/checkbox.e2e-spec.ts

```diff
@@ -0,0 +1,346 @@
+            }
+        });
+
+        it('should check binary checkbox without value', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                for (let i = 2; 4 > i; i++) {
+                    checkIfDisabled(checkboxPage.checkboxWithoutForm, 'ng-reflect-is-disabled', 'false', i);
+                    checkMarkingCheckbox(checkboxPage.checkboxWithoutForm, i);
+                }
+
+                checkFocusState(checkboxPage.checkboxWithoutForm, 2);
+                checkHoverState(checkboxPage.checkboxWithoutForm, 2);
+            }
+        });
+
+        it('should check disabled checkbox', () => {
+            // TODO: Fix for Safari
+            if (browser.capabilities.browserName === 'Safari') {
+                console.log('skip');
+            } else {
+                webDriver.waitElementToBePresentInDOM(checkboxPage.disabledCheckboxWithoutForm);
+                webDriver.scrollIntoView(checkboxPage.disabledCheckboxWithoutForm);
+                webDriver.waitForDisplayed(checkboxPage.disabledCheckboxWithoutForm);
+                checkIfDisabled(checkboxPage.disabledCheckboxWithoutForm, 'ng-reflect-is-disabled', 'true');
+            }
+        });
+    });
+
+    describe('Check checkboxes with value property examples', () => {
+        it('should check template driven form', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                for (let i = 0; 2 > i; i++) {
+                    checkIfDisabled(checkboxPage.checkboxWithValue, 'ng-reflect-is-disabled', 'false', i);
+                    checkMarkingCheckbox(checkboxPage.checkboxWithValue, i);
+                }
+                checkHoverState(checkboxPage.checkboxWithValue);
+                checkFocusState(checkboxPage.checkboxWithValue);
+            }
+        });
+        it('should check reactive/model driven form', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                for (let i = 2; 4 > i; i++) {
+                    checkIfDisabled(checkboxPage.checkboxWithValue, 'ng-reflect-is-disabled', 'false', i);
+                    checkMarkingCheckbox(checkboxPage.checkboxWithValue, i);
+                }
+
+                checkHoverState(checkboxPage.checkboxWithValue, 2);
+                checkFocusState(checkboxPage.checkboxWithValue, 2);
+            }
+        });
+    });
+
+    describe('Check Tristate Checkbox With Value Property and Without Value Property', () => {
+        it('should check reactive form', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                for (let i = 0; 8 > i; i++) {
```

Comment:
magic numbers?

Reply:
Some components have lots of examples with exec same locators. So it is two ways to identify them by index or by exact locations. In this case, the index looks more stable (less readable). We will think how to optimize that in the future(by adding the ids or unique attr)